Brokerages/Alpaca/AlpacaBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+
+        /// <summary>
+        /// The market hours database
+        /// </summary>
+        private readonly MarketHoursDatabase _marketHours;
+
+        private readonly Dictionary<Symbol, DateTimeZone> _symbolExchangeTimeZones = new Dictionary<Symbol, DateTimeZone>();
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AlpacaBrokerage"/> class.
+        /// </summary>
+        /// <param name="orderProvider">The order provider.</param>
+        /// <param name="securityProvider">The holdings provider.</param>
+        /// <param name="accountKeyId">The Alpaca api key id</param>
+        /// <param name="secretKey">The api secret key</param>
+        /// <param name="tradingMode">The Alpaca trading mode. paper/live</param>
+        public AlpacaBrokerage(IOrderProvider orderProvider, ISecurityProvider securityProvider, string accountKeyId, string secretKey, string tradingMode)
+            : base("Alpaca Brokerage")
+        {
+            var baseUrl = "api.alpaca.markets";
+            if (tradingMode.Equals("paper")) baseUrl = "paper-" + baseUrl;
+            baseUrl = "https://" + baseUrl;
+
+            _orderProvider = orderProvider;
+            _securityProvider = securityProvider;
+
+            _marketHours = MarketHoursDatabase.FromDataFolder();
+
+            // api client for alpaca
+            _restClient = new RestClient(accountKeyId, secretKey, baseUrl);
+
+            // websocket client for alpaca
+            _sockClient = new SockClient(accountKeyId, secretKey, baseUrl);
+            _sockClient.OnTradeUpdate += OnTradeUpdate;
+            _sockClient.OnError += OnSockClientError;
+
+            // polygon client for alpaca
+            _natsClient = new NatsClient(accountKeyId, baseUrl.Contains("staging"));
+            _natsClient.QuoteReceived += OnQuoteReceived;
+            _natsClient.TradeReceived += OnTradeReceived;
+            _natsClient.OnError += OnNatsClientError;
+        }
+
+        #region IBrokerage implementation
+
+        /// <summary>
+        /// Returns true if we're currently connected to the broker
+        /// </summary>
+        public override bool IsConnected => _isConnected && !_connectionLost;
+
+        /// <summary>
+        /// Connects the client to the broker's remote servers
+        /// </summary>
+        public override void Connect()
+        {
+            if (IsConnected) return;
+
+            _sockClient.ConnectAsync().SynchronouslyAwaitTask();
+            _natsClient.Open();
+
+            _isConnected = true;
+
+            // create new thread to manage disconnections and reconnections
+            _cancellationTokenSource = new CancellationTokenSource();
+            _connectionMonitorThread = new Thread(() =>
+            {
+                try
+                {
+                    while (!_cancellationTokenSource.IsCancellationRequested)
+                    {
+                        Thread.Sleep(10000);
```

Comment:
Maybe we could use some combination of `Thread.Sleep` + using the cancellation token, so we do not wait 10 seconds on this thread